Source/WebKit/NetworkProcess/Downloads/PendingDownload.cpp

```diff
@@ -43,22 +44,53 @@ WTF_MAKE_TZONE_ALLOCATED_IMPL(PendingDownload);
 
 PendingDownload::PendingDownload(IPC::Connection* parentProcessConnection, NetworkLoadParameters&& parameters, DownloadID downloadID, NetworkSession& networkSession, const String& suggestedName, FromDownloadAttribute fromDownloadAttribute, std::optional<WebCore::ProcessIdentifier> webProcessID)
     : m_networkLoad(NetworkLoad::create(*this, WTFMove(parameters), networkSession))
+    , m_downloadID(downloadID)
     , m_parentProcessConnection(parentProcessConnection)
     , m_fromDownloadAttribute(fromDownloadAttribute)
     , m_webProcessID(webProcessID)
 {
-    m_networkLoad->start();
-    m_isAllowedToAskUserForCredentials = parameters.clientCredentialPolicy == ClientCredentialPolicy::MayAskClientForCredentials;
+    relaxAdoptionRequirement();
 
-    m_networkLoad->setPendingDownloadID(downloadID);
-    m_networkLoad->setPendingDownload(*this);
-    m_networkLoad->setSuggestedFilename(suggestedName);
+#if ENABLE(CONTENT_FILTERING)
+    NetworkProcess::setSharedParentalControlsURLFilterIfNecessary();
+#endif
+
+#if HAVE(WEBCONTENTRESTRICTIONS)
+#if HAVE(WEBCONTENTRESTRICTIONS_PATH_SPI)
+    m_urlFilter = ParentalControlsURLFilter::filterWithConfigurationPath();
+#else
+    m_urlFilter = ParentalControlsURLFilter::singleton();
+#endif // HAVE(WEBCONTENTRESTRICTIONS_PATH_SPI)
+#endif // HAVE(WEBCONTENTRESTRICTIONS)
```

Comment:
I don't think we need to address this in this patch, but should downloads be checked against all content filters?

Reply:
One the one hand, it seems like it would be desirable (And my first attempt at this did, in fact, try to run it against ContentFilter and all implementations)

A few of us who discussed concluded:
1 - Actual content analysis of the downloads is not possible in WebKit today, at least in the Cocoa port, because we never see the bytes
2 - Our actual content analysis frameworks that WebKit uses are very targeted to small bits of web content, and not a 100mb PDF file or a 2gb media file, e.g.
3 - Consulting the URL filters solves the task at hand
4 - If we decide we need to analyze the entire file later, there's other strategies that might make way more sense than having WebKit do it at download time.